package.json

```diff
@@ -4,7 +4,11 @@
   "description": "hashmap + constructor function = a hashmap you don't have to check before calling get()",
   "main": "index.js",
   "scripts": {
-    "test": "tape test.js && jsmd readme.md"
+    "build": "rm -rf dist && npm test && npm run compile && cp package.json readme.md dist/",
+    "compile": "npx tsc --project .",
+    "test": "npx tslint --project . && npx tsc --project tsconfig-tests.json && tape test.js && jsmd readme.md",
+    "prepublishOnly": "echo 'Run \\'npm run publish-package\\' instead' && exit 1",
+    "publish-package": "git push && git push --tags && npm run build && npm publish --access=public --ignore-scripts dist"
```

Comment:
This isn't necessary since I use some [handy publish aliases](https://gist.github.com/TehShrike/5f204246f75d3d1dc8a0#file-bash_profile-L34-L38)